src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -440,4 +439,191 @@ private static void _BulkMoveWithWriteBarrier(ref byte destination, ref byte sou
 
 #endif // !MONO
     }
+
+    internal static partial class SpanHelpers
```

Comment:
This is violating coding conventions (the class name does not match the file name) and it will cause confusion.

What would be the best place to land with this? Move all these vectorized helpers (including MemMove) to SpanHelpers?